e2e-testing/cypress/functions/PostFunctions.js

```diff
@@ -0,0 +1,61 @@
+/// <reference types="Cypress" />
+
+import PostLocators from '../locators/PostLocators';
+
+class PostFunctions {
+  constructor() {
+    this.postTitle = 'Automated Title Response';
+    this.postDescription = 'Automated Description Response';
+  }
+  open_post_creation_form() {
+    cy.get('.sidebar__add-post-btn .submit-post-button .mzima-button--primary').click();
+    cy.get(PostLocators.srvyItemBtn).click();
+  }
+
+  fill_required_form_fields() {
+    cy.contains('mat-label', 'Title')
+      .siblings('mat-form-field')
+      .find('input.mat-input-element')
+      .type(this.postTitle);
+    cy.get(PostLocators.descField).type(this.postDescription);
+    cy.get(PostLocators.lngTextField).type('This is an automated long text response');
+    cy.contains('mat-label', 'Number (Decimal) Field')
+      .siblings('mat-form-field')
+      .find('input.mat-input-element')
+      .type(99.9);
+    cy.contains('mat-label', 'Number (Integer) Field')
+      .siblings('mat-form-field')
+      .find('input.mat-input-element')
+      .type(100);
+    // click on date field to open pop up
+    cy.contains('mat-label', 'Date & Time Field')
+      .siblings('mat-form-field')
+      .find('input.mat-input-element')
+      .click();
+    cy.get('.mat-button.mat-stroked-button').click(); // click on the checkmark button to select today's date
+    cy.get(PostLocators.selectFieldBtn).click();
+    cy.get(PostLocators.selectFieldOption1).click();
+    cy.get(PostLocators.radioFieldOption2).click();
+    cy.get(PostLocators.checkboxFieldOption3).click();
+    cy.get(PostLocators.relatedPostField).type(
+      'http://localhost:4200/feed/103214/view?page=1&mode=POST',
```

Comment:
Let's leave the related post field blank.
We'll have a separate task for it, since the process includes searching for an existing post and then linking it.

Reply:
@AmTryingMyBest related post field is a required field on this survey form though 🤔

Reply:
Hadn't noticed that. I'll look into it.